### PR TITLE
feat: [SRM-14703]: Added slo formula type

### PR DESCRIPTION
**Pull request by deepesh-ui**
#### Screenshots

Screenshot 2023-05-05 at 3 55 40 PM

image

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.module.scss [hunk 1]
```diff
@@ -15,11 +15,11 @@
 
 .weightageInput {
   display: flex;
-  justify-content: space-between;
+  // justify-content: center;
```
Comment (harshilbits) [inline src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.module.scss:18]: remove commented code.

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.tsx [hunk 2]
```diff
@@ -267,18 +276,22 @@ export const AddSLOs = (props: AddSLOsProp): JSX.Element => {
     },
     {
       Header: getString('tagsLabel').toUpperCase(),
-      width: '10%',
+      // width: '10%',
```
Comment (harshilbits) [inline src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.tsx:279]: same.

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.tsx [hunk 3]
```diff
@@ -267,18 +276,22 @@ export const AddSLOs = (props: AddSLOsProp): JSX.Element => {
...
       Cell: RenderTags
     },
     {
       accessor: 'sloTargetPercentage',
       Header: getString('cv.slos.target').toUpperCase(),
-      width: '10%',
+      // width: '10%',
```
Comment (harshilbits) [inline src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.tsx:285]: same.

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.tsx [hunk 4]
```diff
@@ -321,12 +338,12 @@ export const AddSLOs = (props: AddSLOsProp): JSX.Element => {
...
         </>
       ),
-      width: '15%',
+      // width: '5%',
       Cell: RenderWeightInput
     },
     {
       id: 'deletSLO',
-      width: '15%',
+      // width: '15%',
```
Comment (harshilbits) [inline src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.tsx:346]: pls check everywhere./

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.tsx [hunk 5]
```diff
@@ -321,12 +338,12 @@ export const AddSLOs = (props: AddSLOsProp): JSX.Element => {
           </Button>
         </>
       ),
-      width: '15%',
+      // width: '5%',
```
Comment (harshilbits) [inline src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.tsx:341]: same

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.tsx [hunk 6]
```diff
@@ -343,36 +360,74 @@ export const AddSLOs = (props: AddSLOsProp): JSX.Element => {
...
-        width={150}
-        loading={dashboardWidgetsLoading}
-        data-testid={'addSlosButton'}
-        variation={ButtonVariation.SECONDARY}
-        text={getString('cv.CompositeSLO.AddSLO')}
-        iconProps={{ name: 'plus' }}
-        onClick={showDrawer}
-        margin={showSLOTableAndMessage ? { bottom: 'large', top: 'large' } : {}}
-      />
+      {/* {showSLOTableAndMessage && <Text>{getString('cv.CompositeSLO.AddSLOMessage')}</Text>} */}
```
Comment (harshilbits) [inline src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.tsx:400]: commented code.

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.tsx [hunk 7]
```diff
@@ -343,36 +356,73 @@ export const AddSLOs = (props: AddSLOsProp): JSX.Element => {
...
   )
   const showErrorState = totalOfSloWeight > 100 || totalOfSloWeight < 100
+  const tabelClassNames = isFormulaWeightedAverage
+    ? {
+        [css.rowFailure]: showErrorState,
+        [css.rowSuccess]: !showErrorState
+      }
+    : {}
+
+  const radioItems: Pick<RadioButtonProps, 'label' | 'value'>[] = useMemo(() => {
```
Comment (praneshg239) [inline src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.tsx:366]: As it does not have any dependency to be within this component, so shall we move this to utils?

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.tsx [hunk 8]
```diff
@@ -343,36 +356,73 @@ export const AddSLOs = (props: AddSLOsProp): JSX.Element => {
...
+          width={150}
+          loading={dashboardWidgetsLoading}
+          data-testid={'addSlosButton'}
+          variation={ButtonVariation.SECONDARY}
+          text={getString('cv.CompositeSLO.AddSLO')}
+          iconProps={{ name: 'plus' }}
+          onClick={showDrawer}
+        />
+        <FormInput.RadioGroup
+          label={'Choose Computation Formula'}
```
Comment (praneshg239) [inline src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.tsx:407]: Please refer `'Choose Computation Formula'` from getString

File: src/modules/85-cv/strings/strings.en.yaml [hunk 9]
```diff
@@ -996,8 +996,13 @@ CompositeSLO:
   SetTarget: Set SLO Target
   ErrorBudgetPolicy: Error Budget Policy
   Weightage: Weightage
+  impactPercentage: Impact Percentage
   NoData: You don’t have any matching SLOs
   NoDataSuggestion: Try changing the time window or clearing your search input.
+  weightedAverage: Weighted Average
```
Comment (praneshg239) [inline src/modules/85-cv/strings/strings.en.yaml:1002]: do we need to have `Weighted Average` or shall we keep it as `Weighted average`?

File: src/modules/85-cv/strings/strings.en.yaml [hunk 10]
```diff
@@ -996,8 +996,13 @@ CompositeSLO:
...
+  weightedAverageSubtext: Use WeightedAverage formula for composite SLOs when SLOs don't impact each other.
```
Comment (praneshg239) [inline src/modules/85-cv/strings/strings.en.yaml:1003]: should we need to have `WeightedAverage` as a property here? Or we can simply keep it as a sentence like `Weighted average`. Same for `LeastPerformance` below

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.tsx (hunk 8 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.tsx:407]: done

File: src/modules/85-cv/strings/strings.en.yaml (hunk 9 above)
Comment (deepesh-ui) [reply src/modules/85-cv/strings/strings.en.yaml:1002]: kept same as mocks

File: src/modules/85-cv/strings/strings.en.yaml (hunk 10 above)
Comment (deepesh-ui) [reply src/modules/85-cv/strings/strings.en.yaml:1003]: kept same as mock

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.tsx (hunk 7 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.tsx:366]: done

**Pull request merged**